app/views/shf_applications/information.html.haml

```diff
@@ -5,17 +5,23 @@
 .entry-content
   - if current_user.member?
     %h2
-      = t('.member.title')
-      = current_user.first_name
-    %p= t('.member.using_the_logo')
-    %p "#{t('.member.upload_this_image')} http://www.sverigeshundforetagare.se"
-    = image_tag('medlem.png')
-    %p
-    %p "#{t('.member.using_the_h_mark')} http://sverigeshundforetagare.se/agare/h-markt-av-sveriges-hundforetagare/"
-    = image_tag('h-marke.png')
-  - else
+    = t('.member.title')
+    = current_user.first_name
+    = render partial: 'unpaid_member'
+    - unless current_user.membership_expire_date.past?
+      %p= t('.member.using_the_logo')
+      %p "#{t('.member.upload_this_image')} http://www.sverigeshundforetagare.se"
+      = image_tag('medlem.png')
+      %p
+      %p "#{t('.member.using_the_h_mark')} http://sverigeshundforetagare.se/agare/h-markt-av-sveriges-hundforetagare/"
+      = image_tag('h-marke.png')
+  - elsif current_user.shf_application.nil? == false && current_user.shf_application.state == 'accepted' && current_user.member? == false
+    = render partial: 'unpaid_new_member'
+  - elsif current_user.shf_application.nil? == false && current_user.shf_application.state != 'accepted' && current_user.member? == false
+    = render partial: 'waiting_for_approval'
+  - if current_user.shf_application.nil?
```

Comment:
See my PR for a refactored version of this file.  Some things to note:

1) If you find your logic looking like this:
```
if test-a and test-b
  ....
elsif test-a and test-c
 ....
```
Then you probably want:
```
if test-a
  if test-b
    ....
  elsif test-c
    ....
```

2. Use model methods whenever possible - these tend to protect against things like calling a method on Nil.  For instance, this will blow up if `membership_expire_date` returns nil (which it can):
```
unless current_user.membership_expire_date.past?
```
Instead:
```
if current_user.membership_current?
```
(if we don't have an appropriate model method then we need to create one).

3. In terms of style, I prefer using white space generously - especially in HAML files where logic is intermixed with rendering code.  White space allows the eye to scan for the logic, without getting caught up in trying to separate the two types of code in your head.

Reply:
Good info :)

Reply:
3 - I know you like whitespace ;)